### PR TITLE
Allow remembering of direct path and auto redirect on login

### PR DESCRIPTION
**Pull request by rawrrencee**
# Description

add ability to remember paths and redirect automatically after login (if valid)
only urls with 'sensei'/'student'/'social' inside the url will be redirected
removed user object check from sendToSocialProfile as it is no longer required

# Changelog:

see commit log

## Checklist:

- [ ] Merged latest develop
- [ ] PR title makes sense

### CODE REVIEW

File: src/layouts/index.js [hunk 1]
```diff
@@ -83,9 +84,15 @@ const Layout = ({ children, location: { pathname, search } }) => {
       if (
         isUserAuthorized ||
         (!isUserAuthorized &&
-          (/^\/sensei(?=\/|$)/i.test(pathname) || /^\/student(?=\/|$)/i.test(pathname)))
-      )
+          (/^\/sensei(?=\/|$)/i.test(pathname) ||
+            /^\/student(?=\/|$)/i.test(pathname) ||
+            /^\/social(?=\/|$)/i.test(pathname)))
+      ) {
+        if (['sensei', 'student', 'social'].some(path => pathname.includes(path))) {
```
Comment (rawrrencee) [inline src/layouts/index.js]: if paths contain 'sensei' / 'student' / 'social' then they will be attempted at a redirect. if the page doesn't exist still, then the redirect will still result in error page (page not found) or redirect back to dashboard (if not authorised)

Comment (rawrrencee) [reply src/layouts/index.js]: actually realise this might be redundant. need to double check but I'll be out today morning! will check again later

Comment (natalieloke) [reply src/layouts/index.js]: ok, do let us know!

**Review by natalieloke — APPROVED** [review]

File: src/layouts/index.js (hunk 1 above)
Comment (rawrrencee) [reply src/layouts/index.js]: i removed in a commit like 2min ago!!

**Pull request merged**